mrtt-ui/src/components/ProjectDetailsForm/ProjectDetailsForm.js

```diff
@@ -38,68 +37,54 @@ function ProjectDetailsForm() {
         })
       )
       .min(1)
-      .typeError('Select at least one item')
+      .typeError('Select at least one country')
   })
   const formOptions = { resolver: yupResolver(validationSchema) }
 
   // get functions to build form with useForm() hook
-  const { register, control, handleSubmit, formState, watch } = useForm(formOptions)
+  const { control, handleSubmit, formState, watch } = useForm(formOptions)
   const { errors } = formState
   const watchHasProjectEndDate = watch('hasProjectEndDate', 'false')
+  const [isSubmitting, setisSubmitting] = useState(false)
+  const [isError, setIsError] = useState(false)
+
+  const onSubmit = async (data) => {
+    setisSubmitting(true)
+    setIsError(false)
+    let preppedData = []
+    const url = 'https://mrtt-api-test.herokuapp.com/api/v2/sites/1/registration_answers'
+
+    if (!data) return
 
-  const onSubmit = (data) => {
     // set up data structure for api
-    // POST to api route
-    console.log('data: ', data)
-  }
+    for (const [key, value] of Object.entries(data)) {
+      // map question ids with keys
+      // eslint-disable-next-line no-prototype-builtins
```

Comment:
I looked at the [docs](https://eslint.org/docs/rules/no-prototype-builtins) for this.
The suggestion is
> To avoid subtle bugs like this, it's better to always call these methods from Object.prototype. For example, foo.hasOwnProperty("bar") should be replaced with Object.prototype.hasOwnProperty.call(foo, "bar").

I don't have a strong opinion on it but it might be better to take that approach to avoid bugs and please the linter. Although it definitely doesn't looks as nice.

Reply:
no problem - just updated!